ros/src/system/autoware_health_checker/src/diag_buffer.cpp

```diff
@@ -0,0 +1,93 @@
+#include <autoware_health_checker/diag_buffer.h>
+
+namespace autoware_health_checker
+{
+    DiagBuffer::DiagBuffer(std::string key, uint8_t type, std::string description, double buffer_length) : type(type), description(description)
+    {
+        key_ = key;
+        buffer_length_ = ros::Duration(buffer_length);
+    }
+
+    DiagBuffer::~DiagBuffer()
+    {
+        
+    }
+
+    void DiagBuffer::addDiag(autoware_system_msgs::DiagnosticStatus status)
+    {
+        std::lock_guard<std::mutex> lock(mtx_);
+        buffer_[status.level].status.push_back(status);
```

Comment:
Over the whole area, you should replace `push_back()` with `emplace_back()`.

Reply:
replaced to emplace_back.